feat(Client): Add a new component for multi-label text classification bulk annotation

keithCuniah opened the pull request:
# Description

Components : 
- [x] TextClassificationBulkAnnotation 
- [x] BulkAnnotation 
- [x] BulkAnnotationForm

Note : 
- I removed the `remove` attribute from the form since we can guess which annotation will be remove only with the `selected` attribute, I will update the doc
- refactoring for later : it could be a nice to have a builder design pattern to manage forms 

What is left : 
- [x] use Vuelidate plugin to manage disable enable submit and cancel buttons (I didn't use vuelidate, since only one computed was enough)
- [x] disable state of the dropdown button
- [x] implement the search in the BulkAnnotation component ( **bug to correct : need to keep the previous selection after search** )
- [x] Style
- [x] unit test
- [x] update documentation

Closes #2267 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] TextClassification  multilabel

**Checklist**

- [x] I have merged the original branch into my forked branch
- [x] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [x] I added comments to my code
- [x] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

Commit:
feat(Client): components TextClassificationBulkAnnotation, BulkAnnotation and BulkAnnotationForm

Commit:
Merge branch 'feature/bulk-annotation-improvement' into feature/new-component-for-multilabel-textclassification-bulk-annotation

Commit:
feat(Client): removed error unused param

Commit:
feat(Client): disable buttons condition

Commit:
feat(Client): style BulkAnnotationForm

Commit:
feat(Client): add search component in BulkAnnotation

Commit:
feat(Client): pass annotations to parent and apply annotation && condition for multilabel

Commit:
feat(Client): test for TextClassificationBulkAnnotation

leiyre committed:
include pen icon

leiyre committed:
pen icon

leiyre committed:
rename

leiyre committed:
add style

leiyre committed:
fix selectOptionSearch

leiyre committed:
rename old component

leiyre commented on frontend/components/commons/header/global-actions/text-classification-bulk-annotation/textClassificationBulkAnnotation.spec.js:
```diff
@@ -0,0 +1,151 @@
+import { shallowMount } from "@vue/test-utils";
```
the whole text-classifier-bulk-annotation folder is better to be outside of commons, in the textClassifier folder

Commit:
feat(Client): prettier warning from github

Commit:
feat(Client): remove console.log

Commit:
feat(Client): code review => change place bulk annotation component

Commit:
feat(Client): keep previous selection after search

Commit:
feat(Client): on update reset cloneInputs and lastSelectedAnnotation

Commit:
feat(Client): sort dropdown by number of records

Commit:
feat(Client): sort dropdown list by record_ids param

Commit:
feat(Client): adapt click on cross or click on label

Commit:
feat(Client): add selectedRecordIds to update number in dropdown items

Commit:
feat(Client): add removed attribute to dropdown item

Commit:
feat(Client): adapt click on cross or click on label

Commit:
feat(Client): notification in case of success/error bulk annotation

Commit:
feat(Client): remove 'return'

Commit:
feat(Client): format labels to send to API

Commit:
feqt5Client-. function factory depending on single/multiLabel

Commit:
feat(Client): adapted props in test unit

Commit:
feat(Client): fix error in case record no have previous annotation

Commit:
feat(Client): removed console.log

Commit:
feat(Client): correct bad synchronisation number of selectedrecords

leiyre committed:
fix styles

Commit:
feat(Client): adapt Unit test props

Commit:
Merge branch 'feature/new-component-for-multilabel-textclassification-bulk-annotation' of https://github.com/recognai/rubrix into feature/new-component-for-multilabel-textclassification-bulk-annotation

Commit:
feat(Client): refacto toast component call

keithCuniah committed:
feat(Client): correct number of records on pills

keithCuniah committed:
feat(Client): refacto => naming and destructuration

keithCuniah committed:
feat(Client): remove unecessary attribute

Commit:
feat(Client): badge custom directive

Commit:
feat(Client): add comment to help using the directive

Commit:
feat(Client): add destructuration in directive

Commit:
feat(Client): close dropdown on click on cancel button

Commit:
feat(Client): refacto naming event on cancel

keithCuniah committed:
feat(Client) feedback toast for bulk annotation (#2336)

# Description
Add feedback on discard/validation for bulk annotation. This concerned
the 3 tasks

ref #2291 

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [x] TextClassification (single and multilabel)
- [x] TokenClassification 
- [x] Text2Text

**Checklist**

- [ ] I have merged the original branch into my forked branch
- N/A I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- N/A I added comments to my code
- [N/A I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- N/A I have added tests that prove my fix is effective or that my
feature works

---------

Co-authored-by: keithCuniah <keithcuniah@gmail.comh>
